### ISSUE TITLE
Support Compose for Desktop

### ISSUE THREAD

**Issue by ebraminio** [Kotlin/kotlin-jupyter#269]
Supporting a GUI kit with jupyter notebook isn't a too crazy idea, https://ipython.readthedocs.io/en/stable/config/eventloops.html as I like to have access to Compose for Desktop https://github.com/JetBrains/compose-jb but that in addition to library additions https://github.com/JetBrains/compose-jb/blob/master/templates/desktop-template/build.gradle.kts needs some plugins for code generation tasks also so I was wondering if the idea of supporting with kotlin-jupyter can be progressed somehow maybe just like other Java toolkits that can be issued even from jshell https://ebraminio.github.io/2018/09/03/use-jshell-for-awt-swing-swt.html or probably not. Thanks!

**Comment by altavir** [comment]
You do not need to have any special support to launch Spring applications from Jupyter. Just create a frame like you do from a regular Java application and it will work. I think I showed that in one of the videos, but I do not remember where.

**Comment by ebraminio** [comment]
Oh I didn't ask about Swing and even have given a link to swing example which I've written and even work but have asked for Compose for Desktop

**Comment by ebraminio** [comment]
Swing example hopefully useful for others,

```Kotlin
import javax.swing.*
import java.awt.GridLayout

UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName())
JFrame().also { frame ->
    frame.title = "Swing Hello World"
    frame.setSize(400, 400)
    frame.layout = GridLayout(0, 2)
    frame.setLocationRelativeTo(null)
    (1..4).map {
        JButton("Button $it").also { button ->
            button.addActionListener({ println("Button $it is clicked") })
        }
    }.forEach(frame::add)
}.setVisible(true)
```

<img width="512" alt="image" src="https://user-images.githubusercontent.com/833473/120078264-e484db80-c0c3-11eb-8db6-7290efbb5c33.png">

**Comment by altavir** [comment]
Compose desktop currently works on top of Swing. So you can launch it in Swing pane. If you want to write composable functions themselves in the jupyter, it probably won't be possible, because it indeed requires an external compiler plugin to be loaded in the host.

**Comment by ebraminio** [comment]
Yes, to say for example write step by step tutorials of learning compose desktop in jupyter, something common for libraries and so, but understandably is hard to do, feel free to close the file if you feel keeping this around won't be productive, I wouldn't mind at least just was believing would be nice to have.

**Comment by alexismanin** [comment]
Is there new developments on this ? 

KotlinConf 2025 schedules a a talk [Compose Prototyping in Kotlin Notebooks](https://kotlinconf.com/schedule/?day=2025-05-22&session=ef86c77e-4285-54bb-bf63-dad98ac97159).

The abstract specifically states (emphasis mine):

> But did you know that Notebooks also **support Compose** and Swing?
> you can set up Notebooks to access existing UI code or **write new from scratch**

So, we might be able to write composable code in notebook cell ? Is there any insight about this ?
So far, I cannot find any documentation or example online on how to do so.
